main/mio.c

```diff
@@ -984,18 +984,21 @@ char *mio_gets (MIO *mio, char *s, size_t size)
 		{
 			size_t i = 0;
 			bool newline = false;
+			size_t pos = mio->impl.mem.pos;
```

Comment:
Could you put comment here?
Like: "Using local vairable here is for optimization.".

We don't have a test mechanism for detecting performance degrading.
So someone can remove your crafted code without understanding the intention for optimization when refactoring.